plugins/scaffolder-backend/src/scaffolder/tasks/DatabaseTaskStore.ts

```diff
@@ -183,20 +183,72 @@ export class DatabaseTaskStore implements TaskStore {
   async list(options: {
     createdBy?: string;
     status?: TaskStatus;
-  }): Promise<{ tasks: SerializedTask[] }> {
-    const queryBuilder = this.db<RawDbTaskRow>('tasks');
+    filters?: {
+      createdBy?: string | string[];
+      status?: TaskStatus | TaskStatus[];
+    };
+    pagination?: {
+      limit?: number;
+      offset?: number;
+    };
+    order?: { order: 'asc' | 'desc'; field: string }[];
+  }): Promise<{ tasks: SerializedTask[]; totalTasks?: number }> {
+    const { createdBy, status, pagination, order, filters } = options ?? {};
+    const queryBuilder = this.db<RawDbTaskRow & { count: number }>('tasks');
+
+    if (createdBy || filters?.createdBy) {
+      const arr: string[] = [];
+      if (createdBy) {
+        arr.push(createdBy);
+      }
+      if (filters?.createdBy) {
+        if (Array.isArray(filters.createdBy)) {
+          arr.push(...filters.createdBy);
+        } else {
+          arr.push(filters.createdBy);
+        }
+      }
+      queryBuilder.whereIn('created_by', [...new Set(arr)]);
+    }
+
+    if (status || filters?.status) {
```

Comment:
I see the pattern of dealing with the filter parameter for `status` or `createdBy`. It would be nice to create some util function which can transform it, and hide that processing noise to the reader of this function. 
Also it will help to create simple unit test for this transformation logic.